### PR TITLE
Implement minmax support for Boolean type

### LINKED ISSUE #5263: parfors argmin on boolean array fails
## Reporting a bug

- [x] I am using the latest released version of Numba (most recent is visible in
 the change log (https://github.com/numba/numba/blob/master/CHANGE_LOG).
- [x] I have included below a minimal working reproducer (if you are unsure how
 to write one see http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports).

This:
```python
import numpy as np
import numba


@numba.njit(parallel=True)
def test(array):
    for i in numba.prange(array.shape[0]):
        np.argmin(array[i])


test(np.array([[1, 0], [0, 0]], dtype=np.bool_))
```
does this:
```
Failed in nopython mode pipeline (step: nopython mode backend)
'Boolean' object has no attribute 'bitwidth'
```
patch to start fixing https://gitter.im/numba/numba?at=5e4a655b0c50da598c164cf1

suspect `argmax` will also fail the same way.

CC: @sleighsoft (reporter)

### PR DESCRIPTION
**Pull request by svrakitin**
Closes #5263.

- [x] Define `Bounded` type for types with minimal and maximal values.
- [x] Implement `Bounded` type for `Boolean` and `Integer` types.
- [x] Add tests for parfors with `min`/`max`/`argmin`/`argmax` on arrays with bool dtype
- [x] Make output equality checks in parfors tests not fail for boolean arrays

### CODE REVIEW

**Comment by esc** [comment]
@svrakitin thanks for submitting this to the Numba tracker, we appreciate your efforts. I have labeled it as 'WIP' for the time being.

**Comment by svrakitin** [comment]
Maybe it makes sense to implement `Bounded` trait for `Float` as well?

File: numba/core/types/abstract.py [hunk 1]
```diff
@@ -451,3 +450,19 @@ def __init__(self, instance_type):
         self.instance_type = instance_type
         super(TypeRef, self).__init__('typeref[{}]'.format(self.instance_type))
 
+
+class Bounded(Type):
+    """
+    For types that have minimal and maximal values
```
Comment (esc) [inline numba/core/types/abstract.py]:
```suggestion
    For types that have minimal and maximal values.
```

File: numba/tests/test_types.py [hunk 2]
```diff
@@ -254,10 +254,13 @@ def test_bitwidth(self):
         """
         All numeric types have bitwidth attribute
```
Comment (esc) [inline numba/tests/test_types.py]: You may need to update this comment.

**Comment by esc** [comment]
I think this will need an additional review from @stuartarchibald and/or @sklam

**Review by esc — DISMISSED** [review]

**Comment by esc** [comment]
@svrakitin thanks for making the changes!

File: numba/core/types/scalars.py [hunk 3]
```diff
@@ -2,18 +2,34 @@
...
 from numba.core import utils
 from numba.core.typeconv import Conversion
 from numba.np import npdatetime_helpers
 
 
-class Boolean(Hashable):
+class Boolean(Hashable, Bounded):    
+    def __init__(self, name):
+        super(Boolean, self).__init__(name)
+        self.bitwidth = 1
```
Comment (sklam) [inline numba/core/types/scalars.py]: i am worry about adding this as a type property. `bitwidth` is an implementation detail. Numba represents it has an 1-bit value and an 8-bits value depending on how it's used.

**Pull request closed without merging**